### PR TITLE
Add article about Stores in Message Queue

### PR DESCRIPTION
**Pull request by nuzil**
## Purpose of this pull request

Pull request adding info about Patch for Stores support for Message Queues.
Its a follow up of this Blog: 
https://community.magento.com/t5/Magento-DevBlog/Patch-for-Magento-Framework-Message-Queue-and-Store-Scopes/ba-p/135567

## Affected DevDocs pages

guides/v2.3/extension-dev-guide/message-queues/message-queues-stores.md 

whatsnew
Added the topic [Message queues and stores](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/message-queues/message-queues-stores.html).

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.3/extension-dev-guide/message-queues/message-queues-stores.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+}
+```
+
+From example before you can see that plugin checks `application_headers` and add there `store_id` parameter. If headers are not exist, then plugin will create them. In this way RabbitMQ message will receive information for which store requrest have to be processed.
+
+### Processing by consumer
+
+[Consumers]:https://devdocs.magento.com/guides/v2.3/config-guide/cli/config-cli-subcommands-queue.html are picking up messages from RabbitMQ queue and processing them. 
+
+So on a step when consumer reads message, extension executes around plugin in:
```
Comment (keharper) [inline guides/v2.3/extension-dev-guide/message-queues/message-queues-stores.md]:
```suggestion
On a step when a consumer reads a message, the extension executes an around plugin, as shown here: 
```

File: guides/v2.3/extension-dev-guide/message-queues/message-queues-stores.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            }
+        }
+        $proceed($message);
+        if (isset($storeId, $currentStoreId) && $storeId !== $currentStoreId) {
+            $this->storeManager->setCurrentStore($currentStoreId);//restore original store value
+        }
+    }
+```
+
+Plugin will check message headers and will set up current store in `storeManager` to received `store_id` value.
```
Comment (keharper) [inline guides/v2.3/extension-dev-guide/message-queues/message-queues-stores.md]:
```suggestion
The plugin checks the message headers and set the current store value in `storeManager` to the received `store_id` value.
```

File: guides/v2.3/extension-dev-guide/message-queues/message-queues-stores.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
+---
+group: php-developer-guide
+title: Architecture of Message Queues
```
Comment (keharper) [inline guides/v2.3/extension-dev-guide/message-queues/message-queues-stores.md]:
```suggestion
title: Message queues and stores
```

**Comment by keharper** [comment]
Also, add a link to this new topic in `_data/toc/php-developer-guide.yml`, somewhere between lines 209 and 229.

File: guides/v2.3/extension-dev-guide/message-queues/message-queues-stores.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+functional_areas:
+  - Services
+---
+
+One of the key features of Message Queues - is a possibility of processing all asynchronous requests only for particular stores.
+
+For support of store scopes extension `Magento_AmqpStore` is responsible. 
+
+{: .bs-callout .bs-callout-warning }
+Support of stores and mentioned extension is only available starting from Magento 2.3.3 version. If you are using version 2.3.1 or 2.3.2, then you can use Magento [PATCH]:https://magento.com/tech-resources/download#download2312
```
[the remark this answers is not in the document]
Comment (nuzil) [reply guides/v2.3/extension-dev-guide/message-queues/message-queues-stores.md]: Should we remove this note then in version of documentation for 2.3.3?

**Comment by nuzil** [comment]
Hi @keharper,
I though you are on vacation this week :-) 
All changes applied

File: guides/v2.3/extension-dev-guide/message-queues/message-queues-stores.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+            }
+        }
+        $proceed($message);
+        if (isset($storeId, $currentStoreId) && $storeId !== $currentStoreId) {
+            $this->storeManager->setCurrentStore($currentStoreId);//restore original store value
+        }
+    }
+```
+
+The plugin checks the message headers and set the current store value in `storeManager` to the received `store_id` value.
```
Comment (keharper) [inline guides/v2.3/extension-dev-guide/message-queues/message-queues-stores.md]:
```suggestion
The plugin checks the message headers and sets the current store value in `storeManager` to the received `store_id` value.
```

**Comment by keharper** [comment]
@nuzil There are a couple of comments that haven't been merged, plus I corrected a typo that I introduced.  I tried to batch commit these comments from GitHub, but I can't write to your repo. Please make these updates and I'll merge them on Friday. Also, don't forget about #4868 :)

**Comment by nuzil** [comment]
Hi @keharper sorry I didnt get notifications :-( 
I aligned changes, tell me if you still need something

**Comment by keharper** [comment]
Thanks @nuzil

**Pull request merged**